### ISSUE TITLE
Uncaught Error: spawn EACCES

### ISSUE THREAD

**Issue by arkhan** [AtomLinter/linter-flake8#62]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.4
**System**: "Funtoo Linux
**Thrown From**: [linter-flake8](https://github.com/AtomLinter/linter-flake8) package, v1.6.0

### Stack Trace

Uncaught Error: spawn EACCES

```
At /usr/share/atom/resources/app.asar/src/buffered-process.js:260

Error: spawn EACCES
  at exports._errnoException (util.js:734:11)
  at ChildProcess.spawn (child_process.js:1145:11)
  at Object.exports.spawn (child_process.js:977:9)
  at BufferedProcess.module.exports.BufferedProcess.spawn (/usr/share/atom/resources/app.asar/src/buffered-process.js:188:44)
  at new BufferedProcess (/usr/share/atom/resources/app.asar/src/buffered-process.js:52:14)
  at /home/arkhan/.atom/packages/linter-flake8/node_modules/atom-linter/lib/helpers.coffee:40:30
  at Object.module.exports.Helpers._exec (/home/arkhan/.atom/packages/linter-flake8/node_modules/atom-linter/lib/helpers.coffee:22:16)
  at Object.module.exports.Helpers.exec (/home/arkhan/.atom/packages/linter-flake8/node_modules/atom-linter/lib/helpers.coffee:13:13)
  at Object.provider.lint (/home/arkhan/.atom/packages/linter-flake8/lib/main.coffee:65:24)
  at Promise.then._this.emitter.emit.linter (/home/arkhan/.atom/packages/linter/lib/linter-registry.coffee:55:22)
  at LinterRegistry.triggerLinter (/home/arkhan/.atom/packages/linter/lib/linter-registry.coffee:54:16)
  at /home/arkhan/.atom/packages/linter/lib/linter-registry.coffee:48:17
  at Array.map (native)
  at /home/arkhan/.atom/packages/linter/lib/linter-registry.coffee:46:27

```

### Commands

```
     -0:20.2.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:19.4.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:19 editor:newline (atom-text-editor.editor.is-focused)
  7x -0:14.1.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:12 core:backspace (atom-text-editor.editor.is-focused)
  6x -0:11.6.0 core:move-right (atom-text-editor.editor.is-focused)
  8x -0:10.6.0 core:select-left (atom-text-editor.editor.is-focused)
     -0:09.6.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:09.1.0 core:move-up (atom-text-editor.editor.is-focused)
     -0:08.5.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:08.2.0 core:move-up (atom-text-editor.editor.is-focused)
     -0:03.1.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
  3x -0:02.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:00.9.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true
  },
  "linter-flake8": {
    "executablePath": "/usr/bin/"
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-plus-python-jedi, v0.3.6
linter, v1.3.1
linter-flake8, v1.6.0
minimap, v4.12.2
odoo-snippets, v0.1.0
project-manager, v1.15.11
python-isort, v0.0.7

# Dev
No dev packages
```

**Comment by Arcanemagus** [comment]
Are you sure `flake8` is executable by the user `atom` is running under?

**Comment by arkhan** [comment]
yes, I execute flake8 under the terminal, and work

**Comment by Arcanemagus** [comment]
Actually it looks like you have configured an `executablePath`, but specified a directory instead of the `flake8` executable. Please specify the full path to `flake8` in there and this should be fixed.

**Comment by badray** [comment]
Just like @Arcanemagus @arkhan please run in terminal `$ which flake8` and place output to `executablePath` setting.

**Comment by badray** [comment]
@arkhan Any feedback pls?

**Comment by jeremyosborne** [comment]
If this helps anyone, I just got this today installing linter-flake8.

Using the settings of linter-flake8, setting just the dir for the `which flake8` executable didn't work, as in this failed (setting within settings page):

Executable Path = /usr/local/bin

but setting the full executable path works just fine:

Executable Path = /usr/local/bin/flake8

No more process spawning errors.

This could turn out to be an X/Y sort of answer, but superstition or not it worked for me.

**Comment by Arcanemagus** [comment]
@jeremyosborne Maybe "path" is the wrong word to use for those not familiar with filesystems, but that is the proper term for the full path of the executable. Putting just `/usr/local/bin` would cause it to try to execute that directory _as_ the `flake8` executable.

Can you think of a better wording that would make that clearer to people not familiar with working with that? Maybe just adding some descriptive text to make it clear that it should include the binary?

**Comment by jeremyosborne** [comment]
@Arcanemagus I think "path" is fine and good, I just misinterpreted it. It was a derpy moment.

What was most confusing to me was the error that was thrown by atom, and in trying to solve the problem I did attempt number one and absent mindedly put the `dirname` equivalent in instead of the executable path.

What would be most helpful is perhaps a more obvious explication of the spawn error. It took me a moment to figure out what was breaking. I couldn't tell if it was a `flake8` problem or a `linter-flake8` problem, or a combination of them both.
